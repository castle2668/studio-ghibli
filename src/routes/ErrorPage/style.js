import styled from "styled-components";

export const ErrorPage = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
`;
