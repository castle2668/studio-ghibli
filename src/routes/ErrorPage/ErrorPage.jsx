import React from "react";
import { useRouteError } from "react-router-dom";

import * as Style from "./style";

const ErrorPage = () => {
  const error = useRouteError();
  console.error(error);

  return (
    <Style.ErrorPage>
      <h1>Oops!</h1>
      <p>Sorry, an unexpected error has occurred.</p>
      <p>
        <i>{error.statusText || error.message}</i>
      </p>
    </Style.ErrorPage>
  );
};

export default ErrorPage;
