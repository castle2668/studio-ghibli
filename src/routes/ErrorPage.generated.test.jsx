import React from "react";
import renderer from "react-test-renderer";

import ErrorPage from "./ErrorPage";

jest.mock("react-router-dom");

const renderTree = (tree) => renderer.create(tree);
describe("<ErrorPage>", () => {
  it("should render component", () => {
    expect(renderTree(<ErrorPage />).toJSON()).toMatchSnapshot();
  });
});
