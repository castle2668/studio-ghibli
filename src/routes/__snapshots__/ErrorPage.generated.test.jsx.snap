// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage> should render component 1`] = `
<div
  id="error-page"
>
  <h1>
    Oops!
  </h1>
  <p>
    Sorry, an unexpected error has occurred.
  </p>
  <p>
    <i />
  </p>
</div>
`;
